Hot fix

qle2 opened the pull request:
This PR includes:

- Updating our `README` to include dependencies, instructions to set up Conda environment and Neo4j, and running the pipeline
- Refactor `models_to_neo4j.py` and `models.py` to `py2neo`'s new changes to their Graph function
- In `misc.py`, create an output folder with the same name if there isn't one
- Update our `mlapp.yml` and `requirements.txt`. These two files help the user sets up the correct Conda environment to run the pipeline. This should work on both Windows and Ubuntu

qle2 committed:
Change the content of mlapp.yml. This yml file contains the prerequisites required to create the mlapp environment

qle2 committed:
Change the content of requirements.txt. This txt file contains all the remaining packages that are essential to the pipeline

qle2 committed:
Fix logical error in misc.py where the code output to a folder that might not exist. Update: If there is no folder to output to, create one with the same name.

qle2 committed:
Remove pip_installs.txt. In requirements.txt, add all the remaining packages that the pipeline needs

qle2 committed:
Add single_model() function to main. This function will run only one model to completion

qle2 committed:
py2neo's 2021 update has changed the way their function, Graph is implemented. This change refactored models_to_neo4j.py to this new update

qle2 committed:
py2neo's 2021 update has changed the way their function, Graph is implemented. This change refactored models.py to this new update

qle2 committed:
Add single_model() to the main function. This function will run only one model to completion

qle2 committed:
Update our README.md with important information such as: Dependencies, how to create the required environment, how to use Neo4j, how to run our pipeline

qle2 committed:
Update Neo4j instructions. In Neo4j's latest edition, GDS is pre-installed in your DBs

qle2 committed:
Update Environment Set Up instructions. The instruction should work on both Windows and Ubuntu

pep8speaks commented:
Hello @qle2! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`core/models.py`](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/core/models.py):

> [Line 146:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/core/models.py#L146): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (137 > 79 characters)

* In the file [`core/name.py`](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/core/name.py):

> [Line 96:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/core/name.py#L96): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (97 > 79 characters)

* In the file [`core/neo4j/models_to_neo4j.py`](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/core/neo4j/models_to_neo4j.py):

> [Line 320:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/core/neo4j/models_to_neo4j.py#L320): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 410:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/core/neo4j/models_to_neo4j.py#L410): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 79 characters)
> [Line 443:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/core/neo4j/models_to_neo4j.py#L443): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)





* In the file [`main.py`](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py):

> [Line 55:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L55): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (127 > 79 characters)
> [Line 63:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L63): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 64:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L64): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (97 > 79 characters)
> [Line 66:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L66): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (99 > 79 characters)
> [Line 67:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L67): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 79 characters)
> [Line 69:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L69): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 79 characters)
> [Line 70:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L70): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 71:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L71): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 72:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L72): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 73:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L73): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 74:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L74): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (109 > 79 characters)
> [Line 76:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L76): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 79 characters)
> [Line 80:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L80): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 79 characters)
> [Line 82:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L82): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 86:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L86): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 79 characters)
> [Line 95:5](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L95): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 98:5](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L98): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 101:5](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L101): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 102:26](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L102): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 102:33](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L102): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 102:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L102): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 113:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L113): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)
> [Line 121:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L121): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 123:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L123): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 124:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L124): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 126:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L126): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)
> [Line 131:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L131): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 137:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L137): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 143:25](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L143): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 143:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L143): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 144:40](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L144): [E114](https://duckduckgo.com/?q=pep8%20E114) indentation is not a multiple of four (comment)
> [Line 144:40](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L144): [E116](https://duckduckgo.com/?q=pep8%20E116) unexpected indentation (comment)
> [Line 144:40](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L144): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 192:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L192): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (135 > 79 characters)
> [Line 193:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L193): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)
> [Line 193:95](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L193): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 196:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L196): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)
> [Line 199:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/9260afec4f11e963c49271707740b712ac52dd64/main.py#L199): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (119 > 79 characters)



##### Comment last updated at 2021-08-02 16:48:03 UTC

qle2 committed:
Add pip install instructions to download the remaining packages

Commit:
Update to add instructions for MySQL

qle2 committed:
Remove unnecessary comments from code when there are similar file names

qle2 committed:
Refactor function names.
- all_models(): Was main(). Run all possible model combination
- some_models(): Run a few exmaple models
- single_model(): Run a single model

Explain the different functions in main.py and what they do. Redo main.py instructions

qle2 committed:
Rename main() to all_models(). Add some_models() which will run only a few example models.

qle2 committed:
Merge remote-tracking branch 'origin/hot_fix' into hot_fix

# Conflicts:
#	README.md

qle2 committed:
Uncomment to_neo4j() function call. Set featurization from mysql to False

Commit:
update to fix bugs with Qsar

Commit:
update README to have instructions for MySql

Commit:
add data to github

Commit:
add data to github

qle2 committed:
Rename `output` folder to `example`. `example` contains example models that users can load in to Neo4j directly using the function load_example_to_neo4j in `main.py`. The `output` folder will contain new future models generated by the user

qle2 committed:
Add function `load_example_to_neo4j`. This function will open zip files in the `example` folder and load their data into Neo4j

qle2 committed:
Create a Table of Contents. Note Pycharm as optional. Create instructions on how to run `main.py` from the command line

qle2 committed:
Rename `all_models` as `multi_model`. Remove `classification` related code from `multi_model`.

qle2 committed:
Remove vestigial comments

qle2 committed:
Rename `Getting Started` section to `Requirements`. Add `Neo4j` download link

qle2 committed:
Give password instructions to creating a Neo4j `DBMS`. Refactor `MySql` to `MySQL`. Redo the `Running main.py` section

qle2 committed:
Refactor `MySql` to `MySQL` in the `Table of Contents`

qle2 committed:
Add a `Back to Top` option at the bottom

qle2 committed:
Give better instructions on how to run `main.py` from the command prompt

qle2 committed:
change function name. load_folder_to_neo4j now points to a folder parameter instead of hard coding example folder

Commit:
I guess I need to commit here

Commit:
I guess I need to commit here

Commit:
Add using_prebuilt_mlkg.md and screenshots for it

Commit:
Add using_prebuilt_mlkg.md and screenshots to go with it
